modules/net-vpc-firewall/main.tf

```diff
@@ -47,7 +47,7 @@ locals {
     if contains(["EGRESS", "INGRESS"], r.direction)
   }
   _named_ranges = merge(
-    can(var.factories_config.cidr_tpl_file) ? yamldecode(file(var.factories_config.cidr_tpl_file)) : {},
+    can(var.factories_config.cidr_tpl_file) ? var.factories_config.cidr_tpl_file != null ? yamldecode(file(var.factories_config.cidr_tpl_file)) : {} : {},
```

Comment:
can you merge the conditions with `&&` instead of adding another ternary?

Reply:
I tried that in #1991 and also ```var.factories_config != null && var.factories_config.cidr_tpl_file != null ? yamldecode(file(var.factories_config.cidr_tpl_file)) : {}``` just now, but in both cases the tests failed. Or do you have something else in mind?

Reply:
```The logical operators in Terraform do not short-circuit, meaning var.foo || var.foo.bar will produce an error message if var.foo is null because both var.foo and var.foo.bar are evaluated.``` I think the same applies for `&&` as well.

Reply:
You right. I always forget logical operators don't short circuit.